### DISCUSSION TITLE
#keeponlyattributesizes doesn't work

### DISCUSSION THREAD

**Discussion by graqu** [maizzle/framework#1102] [Q&A]
Hello, I am struggling with #keeponlyattributesizes. I want my image to inline styles exept width and make it as an attribute. I want to make something like this in my config.cjs file: 
  inlineCSS: {
    applyWidthAttributes: ["img"],
    keepOnlyAttributeSizes: {
      width: ["img"],
    },
  },
but after run it shows me an error: 
✖ Failed to compile template: index.html
⚠ attrStr.split is not a function

when i will delete     keepOnlyAttributeSizes object everything works but i have inline styles and width attribute either - I want to keep only width as an attribute. How I can do it ?

**Comment by cossssmin** [comment]
Hi there, yes there's a note in the docs for this:

> This will only work for elements defined in [styleToAttribute](https://maizzle.com/docs/transformers/inline-css#styletoattribute)

So you need `styleToAttribute`, this should work:

```js
module.exports = {
  inlineCSS: {
    styleToAttribute: {
      width: 'width',
    },
    applyWidthAttributes: ['img'],
	keepOnlyAttributeSizes: {
      width: ['img']
    }
  }
}
```

**Reply by graqu** [reply]
Thank You for response, Unfortunately, when I copypaste this code, error still shows and it doesn't work.

**Reply by cossssmin** [reply]
Hmm, might be a bug that we're not catching in the [inlining test](https://github.com/maizzle/framework/blob/6b1681cb53a4c11df5ce6d4aaa091a71286cded3/test/test-transformers.js#L89), will look into it.

**Comment by cossssmin** [comment]
Merged #1110, should be fixed now in `v4.6.3`.
